subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/SubscriptionDescriptorFileParser.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.maven.generator.io.files.parser;
+
+import uk.gov.justice.domain.subscriptiondescriptor.SubscriptionDescriptor;
+import uk.gov.justice.domain.subscriptiondescriptor.SubscriptionDescriptorDef;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+public class SubscriptionDescriptorFileParser implements FileParser<SubscriptionDescriptor> {
+
+    @Override
+    public Collection<SubscriptionDescriptor> parse(final Path baseDir, final Collection<Path> paths) {
+        return paths.stream()
+                .map(path -> read(baseDir.resolve(path).toAbsolutePath().toString()))
+                .collect(Collectors.toList());
+    }
+
+    private SubscriptionDescriptor read(final String filePath) {
+        final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
```

Comment:
I think we can add the following...then the setters in the classes can be replaced with constructors
```

                .registerModule(new Jdk8Module())
                .registerModule(new ParameterNamesModule(PROPERTIES))
```